kratos/processes/structured_mesh_generator_process.h

```diff
@@ -58,10 +58,14 @@ namespace Kratos
       /// Default constructor is deleted.
       StructuredMeshGeneratorProcess() = delete;
 
-	  /// Constructor to be used. Takes the geometry to be meshed and ModelPart to be filled
-	  StructuredMeshGeneratorProcess(GeometryType& rGeometry, ModelPart& rOutputModelPart, Parameters TheParameters);
+      /// Constructors to be used. They take the geometry to be meshed and ModelPart to be filled. The second constructor is
+      /// provided for the Python interface.
+      StructuredMeshGeneratorProcess(const GeometryType& rGeometry, ModelPart& rOutputModelPart, Parameters& TheParameters);
 
-	  /// The object is not copyable.
+      StructuredMeshGeneratorProcess(GeometryType::Pointer pGeometry, ModelPart& rOutputModelPart, Parameters& TheParameters):
+          StructuredMeshGeneratorProcess(*pGeometry, rOutputModelPart, TheParameters){KRATOS_WATCH(mNumberOfDivisions)};
```

Comment:
This is dangerous!! you are keeping a reference to a geometry which can be removed by smart pointer....